accepted/0000-sle-micro-reboot-action.md

```diff
@@ -0,0 +1,56 @@
+
+If the system was in its default configuration, and we changed the reboot method in bootstrap, we need to disable services that deals with reboot but are not controlled by Suma.
+
+By default, `transactional-update.timer` is configured to automatically reboot the systems each day. The configuration is on `/etc/systemd/system/timers.target.wants/transactional-update.timer`. To avoid undesired reboots in managed systems, this auto reboot service should be disabled.
+
+`rebootmgr` service allows the user to define maintenance windows where the pending reboots should be performed. This also can cause undesired behavior when the system is managed by Suma, so, if we change the reboot method to put Suma in control, we should also disable `rebootmgr` service.
+
+### Use `transactional-update` module to perform the reboot
+
+In the end, whatever is configured for reboot method, Suma should perform reboot actions calling the [reboot](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.transactional_update.html#salt.modules.transactional_update.reboot) function of `transactional_update` salt module. If `systemd` is configured, the system will reboot immediately and Suma will be aware of the reboot. If other method is configured, Suma will only ask for a reboot and there is no guarantee of when this reboot will be performed. If the reboot doesn't take longer than 6 hours the action will be updated as successfully completed in Suma. But, if there is any peculiarity in the reboot method used (such as a specific maintenance window), and the reboot takes longer than 6 hours, the action will be updated as failed in Suma.
+
+This behavior should be documented to make customers aware with the advice either to stay with `systemd` and let Suma decide when to reboot or they should schedule reboot actions only at a time when their configured method allows to reboot.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Create a `suma` reboot method for `transactional-update`. If the user configure the system to use any other method, reboot wouldn't be allowed in Suma to avoid undesired behavior. This alternative would make it necessary to patch `transactional-update` and the only benefit seems to be having a guarantee that Suma only try to perform reboot if it is in control of it.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- With this proposed approach, we are not covering properly potential users with specific reboot methods configured on systems they manage using Suma, and the plan is to only alert them in documentation about possible problems. Is this scenario of using Suma to manage transactional systems with their own reboot methods really relevant? Does it make sense to use Suma to manage the system but doesn't put it in control of the reboot? If so, we will probably need a next version of this feature improving the support of other reboot methods.
```

Comment:
Users may need to user kubernetes reboot method. In that case, suma just delegates it.